src/app/client-view/training-view/training-day/exercise/exercise.component.ts

```diff
@@ -18,7 +20,9 @@ export class ExerciseComponent implements OnInit {
   exercise: Exercise;
   trainingForm: FormGroup;
 
-  constructor(private trainingService: TrainingService) {}
+  max = this.rmService.max;
```

Comment:
Here's the problem - you're trying to access this max property but the method `calc()` is never called so the property is undefined. You should remove the max property and just call `calc()` here (and probably rename it to something more descriptive)
```suggestion
  max = this.rmService.calcMax(weight, reps, RPE);
```

Also, it's best to define the max variable here: `max: number` then set the value later (eg when the form for an exercise is submitted? Otherwise we won't have any data to use).